File: pe162.cpp
#include "all.h"

int main()
{
	ull sum = 0;
	ull n = 16;
	for(ull x=1;x<n;x++)
		for(ull y=1;y<n;y++)
			for(ull z=1;z<n;z++)
				if(x+y+z <= n)
				{
					ull s = x+y+z;
					ull t = fact(n)/fact(x)/fact(y)/fact(z)/fact(n-s);
					t = t*power(13,n-s);
					sum += t;
				}
	
	for(ull x=1;x<n-1;x++)
		for(ull y=1;y<n-1;y++)
			if(x+y<=n-1)
			{
				ull s= x+y;
				ull t = fact(n-1)/fact(n-1-s)/fact(x)/fact(y);
				t = t*pow(14,n-1-s);
				sum -= t;
			}
	
	cout<<sum;
}

File: pe162.py
import math

def ncr(n,r):
    return math.factorial(n)//math.factorial(r)//math.factorial(n-r)

sum =0
n=3
for n in range(3,17):
    for x in range(1,n):
        for y in range(1,n):
            for z in range(1,n):
                if(x+y+z <= n):
                    s = x+y+z
                    t = ncr(n,s)*ncr(s,x)*ncr(y+z,y)*pow(13,n-s)
                    sum += t
            
    for x in range (1,n-1):
        for y in range (1,n-1):
            if(x+y<=n-1):
                s = x+y
                t = ncr(n-1,s)*ncr(s,x)*pow(14,n-1-s)
                sum -= t
            
print(sum)
